### ISSUE TITLE
Backup restore fails

### ISSUE THREAD

**Issue by pgerber** [signalapp/Signal-Android#9302]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Backup fails to restore. Restore starts, progress shows that hundreds of messages have been restored already and then aborts telling me the passphrase was incorrect.

This is the same issue I reported earlier as #9177 but it was closed as duplicate (#9154). However, that issue was fixed in 4.50.4 but this issue is still present in 4.52.4.

### Steps to reproduce

* create backup on device A
* restore backup on device B

**Actual result:** Restore aborts
**Expected result:** Restore succeeds

### Device info (device A, backup source)

**Device:** Samsung Galaxy
**Android version:** 9 (LineageOS 16)
**Signal version:** 4.52.4

### Device info (device B, backup target)

**Device:** Nokia 4.2
**Android version:** 9
**Signal version:** 4.52.4

### Link to debug log
* [backup creation](https://debuglogs.org/fdc39c4bda97d9bd06abca6cd2f2f67b28eafdafec5a92e93905cf9e55b98cb4)
* [backup restore](https://debuglogs.org/9e278764efd5fd482b2801f6b90cff63fdf207be1e29a6487ac6a16e90d1b392)

Relevant section of restore:

```
12-26 11:39:24.945  6488  6488 I RestoreBackupFragment: Backup restore.
12-26 11:39:27.282  6488  6488 I RestoreBackupFragment: Prompt for backup passphrase shown to user.
12-26 11:39:49.640  6488  6488 I AssistStructure: Flattened final assist data: 8128 bytes, containing 2 windows, 28 views
12-26 11:40:28.046  6488  6644 I RestoreBackupFragment: Starting backup restore.
12-26 11:40:28.054  6488  6530 D OpenGLRenderer: endAllActiveAnimators on 0x767bca8900 (RippleDrawable) with handle 0x767b5016e0
12-26 11:40:29.559  6488  6644 V Cursor  : Filling cursor window with start position:0 required position:0
12-26 11:40:29.603  6488  6644 I FullBackupImporter: Ignoring import for statement: CREATE TABLE sqlite_sequence(name,seq)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: null
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: java.io.IOException: File ended early!
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at org.thoughtcrime.securesms.backup.FullBackupImporter$BackupRecordInputStream.readAttachmentTo(FullBackupImporter.java:281)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at org.thoughtcrime.securesms.backup.FullBackupImporter.processAttachment(FullBackupImporter.java:138)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at org.thoughtcrime.securesms.backup.FullBackupImporter.importFile(FullBackupImporter.java:83)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at org.thoughtcrime.securesms.registration.fragments.RestoreBackupFragment$2.doInBackground(RestoreBackupFragment.java:200)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at org.thoughtcrime.securesms.registration.fragments.RestoreBackupFragment$2.doInBackground(RestoreBackupFragment.java:192)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at android.os.AsyncTask$2.call(AsyncTask.java:333)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
12-26 11:40:32.105  6488  6644 W RestoreBackupFragment: 	at java.lang.Thread.run(Thread.java:764)
```

**Comment by johanw666** [comment]
"java.io.IOException: File ended early!": Looks like the file was truncated somehow? Otherwise you could try the special restore patch from @alan-signal in a new build (see https://github.com/signalapp/Signal-Android/tree/alan/backup-restoration).

**Comment by pgerber** [comment]
Thanks @bepaald and @johanw666 for your help. Turned out this is a bug in OnionShare which I used to transfer the file. Apparently, it truncates the file under certain, yet-to-be-determined circumstances.

I'll still leave the ticket open though. I think the fact that Signal tells the user that the passphrase is wrong is still something that should be addressed. Luckily, I had enough message to restore that I noticed that the restore started but for people with very few message the messages will be very confusing.
